### PR TITLE
Jenkins: builld dsc packages beside amd64 binaries

### PR DESCRIPTION
**Pull request by andriiandrieiev**
## Change Summary

- Building source packages beside amd64 binaries by default (new buildCmd default)
- Source packages added to build artifacts
- Source packages are uploaded to reprepro as well

## Types of changes

 
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Component(s) name

Jenkins libs: 
- buildPackage.groovy
- cloneAndBuild.groovy

## Proposed changes

Build and upload source packages (*.dsc + *.tar.gz) beside with binaries (*.deb) to improve reproducibility of assemblages. Added try-exceptions for packages that can not build debian sources.  
## How to test

Use Jenkins to build any vyos package with default buildPackage() call (90% of packages will be applicable).
Tested in local Jenkins+reprepro environment.
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [ ] I have linked this PR to one or more Phabricator Task(s)
- [ ] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: vars/buildPackage.groovy [hunk 1]
```diff
@@ -136,57 +143,105 @@ def call(description=null, pkgList=null, buildCmd=null, buildArm=false, changesP
...
+                        echo "Uploading Artifacts to external repository"
+                        copyArtifacts fingerprintArtifacts: true, projectName: '${JOB_NAME}', selector: specific('${BUILD_NUMBER}')
+
+                        // build up some fancy groovy variables so we do not need to write/copy
+                        // every option over and over again!
+                        def RELEASE = getGitBranchName()
+                        if (getGitBranchName() == "master")
+                            RELEASE = 'current'
+
+                        def VYOS_REPO_PATH = '/home/sentrium/web/dev.packages.vyos.net/public_html/repositories/' + RELEASE
```
Comment (dmbaturin) [inline vars/buildPackage.groovy:175]: Do you think we could move these paths to the environment variable definitions rather than hardcode them?

**Review by dmbaturin — CHANGES_REQUESTED** [review]
Looks good to me, but I'd move some paths from hardcoded variables to environment variables.

**Review by dmbaturin — CHANGES_REQUESTED** [review]
Looks good to me, but I'd move some paths from hardcoded variables to environment variables.

File: vars/buildPackage.groovy (hunk 1 above)
Comment (andriiandrieiev) [reply vars/buildPackage.groovy:175]: I collided with some number of hardcoded solutions while local Jenkins testing.  It's not a single nuance. Hence I propose to move these parts in another PR.
I can remember these hardcoded parts:

- user ids
- jenkins labels
- Address of repository
- repository path
- repository requirements (reprepro, ucron, specific reprepro settings, etc.)
- Repository usage as fact

Also I found some strange behavior in amr64 builds: toggle true/false option in build func call does not work it's always on, and I couldn't build it in Jenkins on arm64 RPI (ISO extraction fails in the end of ISO builds)

Summing up. I think that it's better to do another one PR which will eliminate hardcode and allows any VyOS user to use our Jenkinsfiles easily. Eventually there is should be created/updated corresponding user documentation.

**Comment by andamasov** [comment]
@andriiandrieiev @dmbaturin, so what do we do with this?

File: vars/buildPackage.groovy (hunk 1 above)
Comment (dmbaturin) [reply vars/buildPackage.groovy:175]: Ok, let's merge it as is, but we need to make sure that a temporary solution with hardcoded URLs doesn't become permanent, as it often happens.
